applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -907,27 +907,49 @@ static bool handle_config_event(struct config_event *event)
 	return true;
 }
 
+static void forward_empty_hid_report(struct hids_peripheral *per,
+				     const struct bt_hogp_rep_info *rep,
+				     uint8_t report_id)
+{
+	if (bt_hogp_rep_type(rep) != BT_HIDS_REPORT_TYPE_INPUT) {
+		return;
+	}
+	size_t size = bt_hogp_rep_size(rep);
+
+	/* Release all pressed keys. */
+	uint8_t empty_data[size];
+
+	memset(empty_data, 0, sizeof(empty_data));
+
+	forward_hid_report(per, report_id, empty_data, size);
+}
+
 static void disconnect_peripheral(struct hids_peripheral *per)
 {
 	LOG_INF("Peripheral %p disconnected", (void *)per);
 
-	struct bt_hogp_rep_info *rep = NULL;
-
-	while (NULL != (rep = bt_hogp_rep_next(&per->hogp, rep))) {
-		if (bt_hogp_rep_type(rep) == BT_HIDS_REPORT_TYPE_INPUT) {
-			uint8_t report_id = bt_hogp_rep_id(rep);
-			size_t size = bt_hogp_rep_size(rep);
+	struct bt_hogp_rep_info *rep;
+	uint8_t report_id;
 
+	if (per->hogp.pm == BT_HIDS_PM_BOOT) {
```

Comment:
I think we should use API call instead of directly accessing structure members. Function is named: `bt_hogp_pm_get`